.evergreen/config.yml

```diff
@@ -2235,8 +2231,13 @@ buildvariants:
     auth: "auth"
     ssl: "nossl"
     encryption: "*"
-  display_name: "Encryption ${platform} ${auth} ${ssl}"
-  tasks: *encryption-server-versions
+  display_name: "Encryption ${platform} ${auth} ${ssl} ${encryption}"
+  tasks: &encryption-server-versions
+    - ".latest"
+    - ".5.0"
+    - ".4.4"
+    - ".4.2"
+    - ".4.0"
```

Comment:
Since we don't want to test CSFLE with <6.0 we should use the "rules" field to ensure we're only testing classic fle with 4.0+ and shared lib fle with 6.0+: https://github.com/evergreen-ci/evergreen/wiki/Project-Configuration-Files#the-rules-field

Reply:
I added a rules field but it does not seem to be removing the tasks.

Reply:
Due to a longstanding bug in EVG, the `if` condition needs to have all matrix axis in it:
```
  rules:
    - if:
        platform: "*"
        auth: "*"
        ssl: "*"
        encryption: [ "encryption_with_csfle" ]
      then:...
```

Reply:
It is now working for mac os, but not for the other platforms.

Reply:
The other ones have different axis.